prefect_gitlab/repositories.py

```diff
@@ -94,6 +94,11 @@ class GitLabRepository(ReadableDeploymentStorage):
         default=None,
         description="An optional reference to pin to; can be a branch name or tag.",
     )
+    git_depth: Optional[int] = Field(
+        default=1,
+        description="The number of commits that Git history is truncated to "
+        "during cloning. Use 0 to disable i.e. get all history.",
+    )
```

Comment:
Since you've typed this a`Optional[int]`, you can use `None` as the value where the entire history is fetched. You also can add validation to ensure that a number greater than or equal that one is provided to avoid errors on the `git clone` command.

```suggestion
    git_depth: Optional[int] = Field(
        default=1,
        gte=1,
        description="The number of commits that Git history is truncated to "
        "during cloning. Set to None to fetch the entire history.",
    )
```

Reply:
thanks @desertaxle for the suggestion. I was thinking of being backwards compatible wrt the existing implementation, i.e. if `git_depth` is not specified (i.e. None) then to use by default `--depth 1` as is done currently. Either way, it's fine for me :-)

Reply:
Thanks for considering backward compatibility! The default value will handle backward compatibility because existing block documents will not specify a `git_depth` and get the default value of `1`. Users will need to explicitly opt into this new functionality by providing a value for `git_depth` including `None`. There's a nuance to Blocks where providing `None` and not providing any value result in different behaviors.